client/my-sites/site-settings/form-general.jsx

```diff
@@ -418,6 +427,55 @@ module.exports = React.createClass( {
 		);
 	},
 
+	renderCustomAddress() {
+		if ( ! config.isEnabled( 'upgrades/domain-search' ) ) {
+			return;
+		}
+
+		return (
+			<Button
+				href={ `/domains/add/'${ this.props.site.slug }` }
+				onClick={ this.trackUpgradeClick }
+			>
+				<Gridicon icon="plus" />
+				{ this.translate( 'Add a Custom Address', { context: 'Site address, domain' } ) }
+			</Button>
+		);
+	},
+
+	renderUpgradePlanNudge() {
+		// do not nudge if the site already has a custom domain
+		// @TODO: wait query-site-domains data component
+		if ( siteHasCustomDomain( this.props.site ) ) {
+			return;
+		}
+
+		// render nudge if the site plan is not free
+		if ( ! siteIsFreePlan( this.props.site.plan ) ) {
+			return;
+		}
+
+		const abtestVariant = abtest( ABTEST_NAME );
+
+		if ( abtestVariant !== 'byUpgradePlanNudge' ) {
+			return;
+		}
+
+		const eventName = `${ EVENT_NAME_PREFIX }${ abtestVariant }`;
+		const title = this.translate( 'Get a free Custom Domain' );
+		const desc = this.translate( 'Custom domains are free when you upgrade to a Premium or Business plan' );
+
+		return (
+			<UpgradeNudge
+				title={ title }
+				message={ desc }
+				feature={ CUSTOM_DOMAIN }
```

Comment:
The nudge is pointing to the base `/plans/:site` page instead of the `/plans/compare/custom-domain/:site` page.